### PR TITLE
Update session flags

### PR DESCRIPTION
**Pull request by tstruk**
Update session enc/dec flags depending on what is supported by the command.

### CODE REVIEW

File: src/tss2-esys/esys_iutil.c [hunk 1]
```diff
@@ -552,6 +552,85 @@ iesys_gen_caller_nonces(ESYS_CONTEXT * esys_context)
...
 }
+
+/** Update session attributes.
+ *
+ * In case where command does not support param encryption/decryption
+ * store the original session attributes and update them accordingly.
+ * Return true is command support param encryption.
+ *
+ * @retval TRUE if command support param encryption
+ * @retval FLASE if command does not support param encryption
```
Comment (williamcroberts) [inline src/tss2-esys/esys_iutil.c]: I was questioning this, and my tests show this version failing.

Why did you do this? The purpose was if flag set and command doesn't support it, quietly turn the flag down.

File: src/tss2-esys/esys_iutil.c [hunk 2]
```diff
@@ -552,6 +552,85 @@ iesys_gen_caller_nonces(ESYS_CONTEXT * esys_context)
...
+ */
+static bool
```
Comment (williamcroberts) [inline src/tss2-esys/esys_iutil.c]: return type: void

File: src/tss2-esys/esys_iutil.c [hunk 3]
```diff
@@ -552,6 +552,85 @@ iesys_gen_caller_nonces(ESYS_CONTEXT * esys_context)
...
+iesys_update_session_flags(ESYS_CONTEXT * esys_context,
+                           IESYS_SESSION *rsrc_session)
+{
+    TSS2_RC r = TSS2_RC_SUCCESS;
+    size_t param_size;
+    const uint8_t *param_buffer;
+    bool encrypt = true;
```
Comment (williamcroberts) [inline src/tss2-esys/esys_iutil.c]: no need for this AFAICT

Comment (tstruk) [reply src/tss2-esys/esys_iutil.c]: We do need this. It's no good if we unset the flags, but still encrypt the parameter.

**Pull request merged**